pkg/proc/gdbserial/gdbserver.go

```diff
@@ -523,9 +523,7 @@ func LLDBLaunch(cmd []string, wd string, flags proc.LaunchFlags, debugInfoDirs [
 		process.SysProcAttr = sysProcAttr(foreground)
 	}
 
-	if runtime.GOOS == "darwin" {
```

Comment:
I do not like that you moved the calls to DisableASyncPreemptEnv out of the backends and into `proc.MergeInheritedEnviron`. It is error prone, as evidenced by the fact you made several errors doing it and now it doesn't get called everywhere it should be called.

I think it would be better to leave them where they are but pass an environment to it to mutate.